backend/Dockerfile.buildapi

```diff
@@ -0,0 +1,38 @@
+# 0. Generate client code (go & json) from API protocol buffers
+FROM golang:1.13.1-stretch as generator
+ENV KFP_VERSION 1.0.0
+ENV GRPC_GATEWAY_VERSION v1.9.0
+ENV GO_SWAGGER_VERSION v0.21.0
+ENV GOLANG_PROTOBUF_VERSION v1.4.2
+ENV PROTOCOLBUFFERS_PROTOBUF_VERSION v3.11.3
```

Comment:
A genuine question, how hard is it to use exactly the same versions as before?
It seems generated code changed substantially.

Note, I'm not saying we must use exact versions, just that it's a tradeoff, it will help us move in smaller steps:
1. refactor the process to remove bazel
2. later upgrade the tools we use to generate code

There can be risk in 2, so depending on efforts, we may choose to do 1&2 together or 1, 2 in separate PRs.

Reply:
> There can be risk in 2, so depending on efforts, we may choose to do 1&2 together or 1, 2 in separate PRs.

I think we should go for separate PR:s to make it easier to review and test. The reason for the current versions used in the PR are due to that I base it upon the work in https://github.com/kubeflow/pipelines/pull/4393 and I honestly don't get which versions are used currently when we generate the API. Have started to dig into how Bazel works and where it pulls the tools from but so far no luck. Would be awesome if you could point me in the right directions for where I can find the exact versions used today. 

> A genuine question, how hard is it to use exactly the same versions as before?

If we can figure out what is used today, I think it should be easy enough to fix.

Reply:
OK, let me take a look too

Reply:
Did you have any success with understanding the versions used today @Bobgy?

Reply:
@NikeNano I believe versions pulled from bazel are defined in https://github.com/kubeflow/pipelines/blob/master/WORKSPACE.

Reply:
Thanks